### PR TITLE
Some changes in controllers

### PR DESCRIPTION
**Pull request by Lusine0101**

### CODE REVIEW

File: Backend/app/controllers/event_bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+class EventBookingsController < ApplicationController
+  before_action :set_event_booking, only: [:show, :update, :destroy]
+   before_action :set_user,only:[:index, :create]
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/event_bookings_controller.rb]: fix the indentation please

File: Backend/app/controllers/event_bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end
+    # Use callbacks to share common setup or constraints between actions.
+    def set_event_booking
+      @event_booking = EventBooking.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def event_booking_params
+      params.fetch(:event_booking, {})
+      # params.require(:room_booking).permit(:available_seats, :starts_at, :ends_at, :room_id)
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/event_bookings_controller.rb]: if you don't need it, please delete it

File: Backend/app/controllers/event_bookings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+  # DELETE /event_bookings/1
+  def destroy
+    @event_booking.destroy
+  end
+
+  private
+   def set_user
+      @user=current_user
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/event_bookings_controller.rb]: styling

File: Backend/app/controllers/rooms_controller.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
+class RoomsController < ApplicationController
+   before_action :set_room, only: [:show, :update, :destroy, :createAvailability, :updateAvailability, :index_availabilities]
+       before_action :authenticate_user!, except: [ :index, :create, :index_availabilities]
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/rooms_controller.rb]: why skip authentication? also fix the styling

File: Backend/app/controllers/rooms_controller.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+   #POST  /room/room_id/
+  def createAvailability
+  
+    @availability = @room.availabilities.new(availability_params)
+   # @availaility.room_id = room.id
+    @availability.save
+    
+  end
+  
+  def updateAvailability
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/rooms_controller.rb]: camel case?!

File: Backend/app/controllers/rooms_controller.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  # PATCH/PUT /rooms/1
+  def update
+    if @room.update(room_params)
+      render json: @room
+    else
+      render json: @room.errors, status: :unprocessable_entity
+    end
+  end
+   #POST  /room/room_id/
+  def createAvailability
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/rooms_controller.rb]: camel case?

File: Backend/app/models/room.rb [hunk 7]
```diff
@@ -1,10 +1,10 @@
 class Room < ApplicationRecord 
 	has_many :availabilities
 	has_many :room_bookings
-	enum room_type: ["conference", "class room"]
```
Comment (a-monemian-vineti) [inline Backend/app/models/room.rb:4]: use :symbols instead of strings

File: Backend/app/models/room_booking.rb [hunk 8]
```diff
@@ -4,5 +4,23 @@ class RoomBooking < ApplicationRecord
   has_one :event
   validates_associated :user, :room, :event
   validates :user_id, :room_id, :starts_at, :ends_at, presence: true 
+    validate :end_date_after_start_date
```
Comment (a-monemian-vineti) [inline Backend/app/models/room_booking.rb]: fix the indentation

File: Backend/app/models/room_booking.rb [hunk 9]
```diff
@@ -4,5 +4,23 @@ class RoomBooking < ApplicationRecord
...
+
+  private
+  def end_date_after_start_date
```
Comment (a-monemian-vineti) [inline Backend/app/models/room_booking.rb]: better use the column names: ends_at_after_starts_at

File: Backend/app/models/room_booking.rb [hunk 10]
```diff
@@ -4,5 +4,23 @@ class RoomBooking < ApplicationRecord
...
+    return if ends_at.blank? || starts_at.blank?
+
+    if ends_at < starts_at
+      errors.add(:ends_at, "must be after the start date")
+    end
+ end
+
+   def starting_time
```
Comment (a-monemian-vineti) [inline Backend/app/models/room_booking.rb]: fix the indentation, and why do we need this?

File: Backend/config/initializers/devise.rb [hunk 11]
```diff
@@ -318,5 +320,8 @@
     jwt.revocation_requests = [
                           ['DELETE', %r{^/logout$}]
                         ]
+   jwt.expiration_time = 2.weeks.to_i
```
Comment (a-monemian-vineti) [inline Backend/config/initializers/devise.rb]: indentation

File: Backend/config/routes.rb [hunk 12]
```diff
@@ -1,4 +1,45 @@
 Rails.application.routes.draw do
```
Comment (a-monemian-vineti) [inline Backend/config/routes.rb]: If you don't need them, please remove the commented lines

File: Backend/db/migrate/20210515093447_change_type_to_be_integer_rooms.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+class ChangeTypeToBeIntegerRooms < ActiveRecord::Migration[6.1]
+  def change
+  	  	  change_column :rooms, :room_type, :integer, default: 0
```
Comment (a-monemian-vineti) [inline Backend/db/migrate/20210515093447_change_type_to_be_integer_rooms.rb]: indentation

File: Backend/spec/requests/event_bookings_spec.rb [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+
+  describe "GET /show" do
+    it "renders a successful response" do
+      event_booking = EventBooking.create! valid_attributes
+      get event_booking_url(event_booking), as: :json
+      expect(response).to be_successful
+    end
+  end
+
+  # describe "POST /create" do
```
Comment (a-monemian-vineti) [inline Backend/spec/requests/event_bookings_spec.rb]: remove instead of comment

File: Backend/spec/requests/users_spec.rb [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+
+  describe "GET /show" do
+    it "renders a successful response" do
+      user = User.create! valid_attributes
+      get user_url(user), as: :json
+      expect(response).to be_successful
+    end
+  end
+
+  # describe "POST /create" do
```
Comment (a-monemian-vineti) [inline Backend/spec/requests/users_spec.rb]: remove instead of commenting

File: Backend/spec/routing/events_routing_spec.rb [hunk 16]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
```
Comment (a-monemian-vineti) [inline Backend/spec/routing/events_routing_spec.rb]: remove the file if you have no tests

File: Backend/spec/routing/users_routing_spec.rb [hunk 17]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
```
Comment (a-monemian-vineti) [inline Backend/spec/routing/users_routing_spec.rb]: remove the file if you have no tests

**Review by a-monemian-vineti — CHANGES_REQUESTED** [review]

File: Backend/app/models/room_booking.rb (hunk 10 above)
Comment (Lusine0101) [reply Backend/app/models/room_booking.rb]: to convert the time and check if (starting_time<ending_time)

File: Backend/app/controllers/events_controller.rb [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+  end
+
+  # GET /events/1
+  def show
+    render json: @event
+  end
+
+  # POST /events
+  def create
+    if @room_booking.event.blank?
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/events_controller.rb]: This is definitely not the right indentation.

File: Backend/app/controllers/events_controller.rb [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    @event = Event.new(event_params)
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/events_controller.rb:24]: I don't think you need `@room_booking`, you can just also permit room_booking_id in the event params and have everything there. And I may have missed something. What is event_type? If it's always gonna be the same as room_type, we don't need it.

File: Backend/app/controllers/events_controller.rb [hunk 20]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show update destroy]
+  before_action :set_user, only: %i[index create]
+  before_action :set_room_booking, only: %i[create]
+ 
+  # GET /events
+  def index
+    @events = Event.all
```
Comment (a-monemian-vineti) [inline Backend/app/controllers/events_controller.rb:10]: Indentation

**Review by a-monemian-vineti — CHANGES_REQUESTED** [review]
Can you please add a new entry in https://github.com/VinetiInternship2021/final-project-URS/blob/main/.github/workflows/ruby.yml to run rubocop for the PRs? the command would be something like this: `bundle exec rubocop --fail-level A`

File: Backend/app/models/room_booking.rb (hunk 10 above)
Comment (a-monemian-vineti) [reply Backend/app/models/room_booking.rb]: Why when do we need that conversion? Is it something that can be done in serializer? You can also configure Rails to format all of time and dates automatically.

File: Backend/app/controllers/events_controller.rb (hunk 19 above)
Comment (Lusine0101) [reply Backend/app/controllers/events_controller.rb:24]: First of all, thank you for your time.
I thought event type means if this event gonna be a conference or lecture? Am I understood wrong?

Comment (a-monemian-vineti) [reply Backend/app/controllers/events_controller.rb:24]: no problem, that's what I'm here for :)
Makes sense, but isn't room type class/lecture hall, and event type lecture/conference which could be in any type of room?

Comment (Lusine0101) [reply Backend/app/controllers/events_controller.rb:24]: I think they're the same room_type and event_type, and that's why I build that way, but if event_type must be changed from UI- so what will be written there?

**Review by TigranMS — APPROVED** [review]

**Review by ElenAvanesyan — APPROVED** [review]

**Review by a-monemian-vineti — APPROVED** [review]

**Pull request merged**